### PR TITLE
Correct size for empty struct element in debug info.

### PR DESCRIPTION
**Pull request by adam-yang**

### CODE REVIEW

File: lib/HLSL/HLModule.cpp [hunk 1]
```diff
@@ -1220,6 +1236,13 @@ void HLModule::CreateElementGlobalVariableDebugInfo(
 
   DIType *DITy = DIGV->getType().resolve(EmptyMap);
   DIScope *DITyScope = DITy->getScope().resolve(EmptyMap);
+
+  // If element size is greater than base size for some reason (for example: 
+  // empty struct), just cap it at that size.
+  unsigned compositeSize = GetCompositeTypeSize(DITy);
```
Comment (MrTrillian) [inline lib/HLSL/HLModule.cpp]: Could we have a fix that's a bit more targeted to the empty struct case? I feel like this should be an assert in the general case and not just silently truncate.

Comment (MrTrillian) [reply lib/HLSL/HLModule.cpp]: Also, this truncation logic is incorrect if `offsetInBits > 0`

**Comment by MrTrillian** [comment]
Did you verify if local variables are subject to the same issue? We should have a test, too.

File: lib/HLSL/HLModule.cpp (hunk 1 above)
Comment (adam-yang) [reply lib/HLSL/HLModule.cpp]: I don't expect the offset in bits to be > 0. Added an assert.

**Comment by adam-yang** [comment]
Local variables are fine. Empty struct local variables don't show up in the debug info.

**Pull request merged**